src/intervalmap.rs

```diff
@@ -0,0 +1,1524 @@
+
+        while !self.node_ref(x, Node::is_sentinel) {
+            y = x;
+            if self.node_ref(z, Node::interval) == self.node_ref(y, Node::interval) {
+                let zval = self.node_mut(z, Node::take_value);
+                let old_value = self.node_mut(y, Node::set_value(zval));
+                return Some(old_value);
+            }
+            if self.node_ref(z, Node::interval) < self.node_ref(x, Node::interval) {
+                x = self.node_ref(x, Node::left);
+            } else {
+                x = self.node_ref(x, Node::right);
+            }
+        }
+        self.node_mut(z, Node::set_parent(y));
+        if self.node_ref(y, Node::is_sentinel) {
+            self.root = z;
+        } else {
+            if self.node_ref(z, Node::interval) < self.node_ref(y, Node::interval) {
+                self.node_mut(y, Node::set_left(z));
+            } else {
+                self.node_mut(y, Node::set_right(z));
+            }
+            self.update_max_bottom_up(y);
+        }
+        self.node_mut(z, Node::set_color(Color::Red));
+
+        self.insert_fixup(z);
+
+        self.len = self.len.wrapping_add(1);
+        None
+    }
+
+    /// Remove a node from the tree.
+    fn remove_inner(&mut self, z: NodeIndex<Ix>) {
+        let mut y = z;
+        let mut y_orig_color = self.node_ref(y, Node::color);
+        let x;
+        if self.left_ref(z, Node::is_sentinel) {
+            x = self.node_ref(z, Node::right);
+            self.transplant(z, x);
+            self.update_max_bottom_up(self.node_ref(z, Node::parent));
+        } else if self.right_ref(z, Node::is_sentinel) {
+            x = self.node_ref(z, Node::left);
+            self.transplant(z, x);
+            self.update_max_bottom_up(self.node_ref(z, Node::parent));
+        } else {
+            y = self.tree_minimum(self.node_ref(z, Node::right));
+            let mut p = y;
+            y_orig_color = self.node_ref(y, Node::color);
+            x = self.node_ref(y, Node::right);
+            if self.node_ref(y, Node::parent) == z {
+                self.node_mut(x, Node::set_parent(y));
+            } else {
+                self.transplant(y, x);
+                p = self.node_ref(y, Node::parent);
+                self.node_mut(y, Node::set_right(self.node_ref(z, Node::right)));
+                self.right_mut(y, Node::set_parent(y));
+            }
+            self.transplant(z, y);
+            self.node_mut(y, Node::set_left(self.node_ref(z, Node::left)));
+            self.left_mut(y, Node::set_parent(y));
+            self.node_mut(y, Node::set_color(self.node_ref(z, Node::color)));
+
+            self.update_max_bottom_up(p);
+        }
+
+        if matches!(y_orig_color, Color::Black) {
+            self.remove_fixup(x);
+        }
+
+        self.len = self.len.wrapping_sub(1);
+    }
+
+    /// Find all intervals in the map that overlaps with the given interval.
+    #[cfg(interval_tree_find_overlap_ordered)]
```

Comment:
We should expose this to user intead of hiding it behind a feature flg.

Reply:
It's ok.